datacube/utils/generic.py

```diff
@@ -20,7 +20,9 @@
 ]
 
 
-def map_with_lookahead(it, if_one=None, if_many=None) -> Generator:
+def map_with_lookahead(
+    it: Iterable, if_one: Callable | None = None, if_many: Callable | None = None
+) -> Generator:
     """
```

Comment:
Seems like we should be able to tighten this signature with generics - something along the lines of:

```
def map_with_lookahead(
    it: Iterable[T1], if_one: Callable[[T1], T2] | None = None, if_many: Callable[[T1], T2] | None = None
) -> Generator[T2]:
```

(with T1 and T2 as TypeVars)

Reply:
Fixed.